changelogs/current.yaml

```diff
@@ -126,6 +129,9 @@ bug_fixes:
 - area: tls
   change: |
     fixed a bug where, when runtime guard ``envoy.reloadable_features.tls_async_cert_validation`` is set to false, the wrong TLS alerts would sometimes be sent in response to certificate validation failures.
+- area: oauth
+  change: |
+    fixed a bug when passthrough header was matched, envoy would always remove the authorization header.
```

Comment:
Is this a bug fix or behavior change? And maybe this will break some exist users? 🤔

Reply:
I think it is a bug fix:
https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/oauth2_filter.html#notes

> pass_through_matcher provides an interface for users to provide specific header matching criteria such that, when applicable, the OAuth flow is **entirely** skipped. When this occurs, the oauth_success metric is still incremented.

Reply:
Sounds like the docs there will need to be updated with this PR since we're no longer incrementing success

While the behavior in this PR seems more correct than before, it does change the behavior slightly and might surprise people. Can we add a runtime check to this change to ensure that people have a way out in case they were relying on this behavior (be it the stats or the sanization)? https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#runtime-guarding